src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,210 @@ type Props = {
-      showExampleCommit !== nextProps.showExampleCommit
+      showExampleCommit !== nextProps.showExampleCommit ||
+      nextState.isLoading !== this.state.isLoading
+    );
+  }
+
+  async fetchProguardMappingFiles(query: string): Promise<Array<DebugFile>> {
+    const {api, organization, project} = this.props;
+    try {
+      const proguardMappingFiles = await api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/files/dsyms/`,
+        {
+          method: 'GET',
+          query: {
+            query,
+            file_formats: 'proguard',
+          },
+        }
+      );
+      return proguardMappingFiles;
+    } catch (error) {
+      Sentry.captureException(error);
+      // do nothing, the UI will not display extra error details
+      return [];
+    }
+  }
+
+  isDataMinified(str: string | null) {
+    if (!str) {
+      return false;
+    }
+
+    return !![...str.matchAll(MATCH_MINIFIED_DATA_REGEX)].length;
+  }
+
+  hasThreadOrExceptionMinifiedFrameData(event: Event, bestThread?: Thread) {
+    if (!bestThread) {
+      const exceptionValues: Array<ExceptionValue> =
+        event.entries?.find(e => e.type === EntryType.EXCEPTION)?.data?.values ?? [];
+
+      return !!exceptionValues.find(exceptionValue =>
+        exceptionValue.stacktrace?.frames?.find(frame =>
+          this.isDataMinified(frame.module)
+        )
+      );
+    }
+
+    const threadExceptionValues = getThreadException(event, bestThread)?.values;
+
+    return !!(threadExceptionValues
+      ? threadExceptionValues.find(threadExceptionValue =>
+          threadExceptionValue.stacktrace?.frames?.find(frame =>
+            this.isDataMinified(frame.module)
+          )
+        )
+      : bestThread?.stacktrace?.frames?.find(frame => this.isDataMinified(frame.module)));
+  }
+
+  async checkProGuardError() {
+    const {event, isShare} = this.props;
+
+    if (!event || event.platform !== 'java') {
+      this.setState({isLoading: false});
+      return;
+    }
+
+    const hasEventErrorsProGuardMissingMapping = event.errors?.find(
+      error => error.type === 'proguard_missing_mapping'
     );
+
+    if (hasEventErrorsProGuardMissingMapping) {
+      this.setState({isLoading: false});
+      return;
+    }
+
+    const proGuardErrors: ProGuardErrors = [];
+
+    const debugImages = event.entries?.find(e => e.type === EntryType.DEBUGMETA)?.data
+      .images as undefined | Array<Image>;
+
+    // When debugImages contains a 'proguard' entry, it must always be only one entry
+    const proGuardImage = debugImages?.find(
+      debugImage => debugImage?.type === 'proguard'
+    );
+
+    const proGuardImageUuid = proGuardImage?.uuid;
+
+    // If an entry is of type 'proguard' and has 'uuid',
+    // it means that the Sentry Gradle plugin has been executed,
+    // otherwise the proguard id wouldn't be in the event.
+    // But maybe it failed to upload the mappings file
+    if (defined(proGuardImageUuid)) {
+      if (isShare) {
```

Comment:
This seems like it could be done on line 156 with the other early return conditions.

Reply:
I thought about it too, but in case of non debug image, the code could still check for minified stack traces, adding maybe the error message `''Some frames appear to be minified. Did you configure the [plugin]?''`.